core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -230,14 +231,27 @@ private JdbcRules() {
   public static List<RelOptRule> rules(JdbcConvention out,
       RelBuilderFactory relBuilderFactory) {
     final ImmutableList.Builder<RelOptRule> b = ImmutableList.builder();
+    b.add(JdbcToEnumerableConverterRule.create(out)
```

Comment:
I don't think that it will be optimal, the main difference is in the presence of `withInTrait(in)` call in a builder chain inside the lambda. Such refactoring will just increase the code complexity.